### PR TITLE
Add Details Screen

### PR DESCRIPTION
**Pull request by MaximBazarov**
# Movie Details Screen
Now I going to add simple details screen, same story as before, UI + Props as the only dependency

### CODE REVIEW

File: Popular Movies/MovieDetailsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
+import UIKit
+
+struct MovieDetailsProps {
```
Comment (oleksii-demedetskyi) [inline Popular Movies/MovieDetailsViewController.swift]: All props should live in `Service.Props` namespace

File: Popular Movies/MovieDetailsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            render()
+        }
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        render()
+    }
+    
+    private func render() {
+        guard let props = props else { return }
```
Comment (oleksii-demedetskyi) [inline Popular Movies/MovieDetailsViewController.swift]: Optional props is a bad pattern. It should be default props from start

File: Popular Movies/MovieDetailsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        nameLabel.text = props.name
+        overviewLabel.text = props.overview
+        let year = Calendar.current.component(.year, from: props.released)
```
Comment (oleksii-demedetskyi) [inline Popular Movies/MovieDetailsViewController.swift]: Date -> String conversion need to be tested, so move it to free function

File: Popular Movies/PopularMoviesViewController.swift [hunk 4]
```diff
@@ -65,7 +65,23 @@ class PopularMoviesViewController: UIViewController {
         return movies.count
     }
     
+    private var selectedMovie: MovieDetailsProps?
     private func selectMovie(at index: Int) {
+        selectedMovie = movie(at: index).map{ movie in
+            return MovieDetailsProps(
```
Comment (oleksii-demedetskyi) [inline Popular Movies/PopularMoviesViewController.swift]: Only `Preseneter` can create props.

**Comment by MaximBazarov** [comment]
This was a huge refactoring, mostly because I was wanted to cut the corners and showcase the cell selection, but now it's even better, there is no real data still, but the whole UI is ready, now let's download the images.

**Comment by MaximBazarov** [comment]
I use `Nuke` as a pod to download images, and of course, it's a dependency (!) and downloading "knowledge" inside the ViewController (!) some of you might have a bad gut feeling about that. But this is perfectly fine, as I said, Components are allowed to have side-effects, this shared component stays only inside the UI, doesn't have an impact on business logic, so it's totally fine to have it like that.

**Pull request merged**